src/stackdriver-nozzle/firehose/client.go

```diff
@@ -25,30 +30,38 @@ func NewClient(apiAddress, username, password string, skipSSLValidation bool) Cl
 			SkipSslValidation: skipSSLValidation}}
 }
 
-func (fc *client) StartListening(nozzle logging.Logging) error {
-	cfClient := cfclient.NewClient(fc.cfConfig)
+func (c *client) StartListening(fh FirehoseHandler) error {
+	cfConfig := &cfclient.Config{
+		ApiAddress:        c.cfConfig.ApiAddress,
+		Username:          c.cfConfig.Username,
+		Password:          c.cfConfig.Password,
+		SkipSslValidation: c.cfConfig.SkipSslValidation}
+	cfClient := cfclient.NewClient(cfConfig)
 
-	cachingClient := caching.NewCachingEmpty()
+	cfConsumer := consumer.New(
+		cfClient.Endpoint.DopplerEndpoint,
+		&tls.Config{InsecureSkipVerify: c.cfConfig.SkipSslValidation},
+		nil)
 
-	eventRouter := eventRouting.NewEventRouting(cachingClient, nozzle)
-
-	//err := eventRouter.SetupEventRouting("LogMessage,ValueMetric,HttpStartStop,CounterEvent,Error,ContainerMetric")
-	err := eventRouter.SetupEventRouting("HttpStartStop,ValueMetric")
-	if err != nil {
-		return err
-	}
+	refresher := CfClientTokenRefresh{cfClient: cfClient}
+	cfConsumer.SetIdleTimeout(time.Duration(30) * time.Second)
+	cfConsumer.RefreshTokenFrom(&refresher)
+	messages, errs := cfConsumer.Firehose("test", "")
```

Comment:
are these needed? looks like they're the subscription id, auth token. Should they be taken as config options?